docs/proposals/202012_receive_split.md

```diff
@@ -8,30 +8,94 @@ status: accepted
+
+In comparison to previous proposal (as mentioned in [alternatives](#previous-proposal-separate-receive-route-command) we have big adventages:
+
+1. We can reduce number of components in Thanos system, we can reuse similar component flags and documentation. Users has to learn about one less command and in result Thanos design is much more approachable. Less components mean less maintainance, code and other implicit duties: Separate changelogs, issue confusions, boilerplates, etc.
+2. Allow consistent pattern with Query. We don't have separate StorAPI component for proxying, we have that baked into Querier. This has been proven to be flexible and understandable, so I would like to propose similar pattern in Receive.
+3. This is more future proof for potential advanced cases like chain of routers -> receives -> routers -> receives for federated writes.
+
+### Open Questions
+
+* Buffering the incoming requests.
+
+### Plan
+
+* Receive without `--receive.hashrings` does not forward or replicate requests, it routes straight to multi-tsdb.
+* Receive without ` --receive.local-endpoint` will assume that no storage is needed, so will skip creating any resources for multi TSDB.
+* Add changes to the documentation (it's simplistic now). Mention two modes.
+
+
+(Inherited from previous proposal, no idea what it is about)
+### Alternative Solutions
+
+#### Previous Proposal: Separate receive-route command
+
 1. Split the receive component into receive-route and receive (and ensure ease of resharding events).
-2. Evaluate any effects on performance by simulating scenarios and collecting and analyzing metrics.
-3. Use consistent hashing to avoid reshuffling time series after resharding events. The exact consistent hashing mechanism to be used needs some further research.
+1. Evaluate any effects on performance by simulating scenarios and collecting and analyzing metrics.
+1. Use consistent hashing to avoid reshuffling time series after resharding events. The exact consistent hashing mechanism to be used needs some further research.
+1. Migration: We document how the new architecture can be set up to have the same general deployment of the old architecture. (We run router and receiver on the same node).
+
+*Pros:*
+
+1. Splitting functionality can be used to make resharding events easier.
+2. Right now the receiver handles ingestion, routing and writing, thus leading to too many responsibilities in a single process.
+   This also makes the receiver more difficult to operate and understand for Thanos users.
+   Splitting the receiver component into two different components could potentially have the following benefits:
+   1. Resharding events become faster and cause no downtime in ingestion.
+   2. Deployment becomes easier to understand for Thanos users.
+   3. Each component consists of less code.
+
+*Cons:*
 
-### Drawbacks of the project
 There is no possible way to have a single-process receiver. The user must have a router + a receiver running.
```

Comment:
If this is the only con, then there are simple solutions to solve this.

Reply:
This is left overall, sorry, will be more explicit

Reply:
I clarified, could you double check?